include/netlicensing/validation_parameters.h

```diff
@@ -41,11 +50,11 @@ namespace netlicensing {
     *            1000 characters.
     */
     void setLicenseeName(const String_t& licenseeName) {
-      licenseeName_i = licenseeName;
+        licenseeParameters_i[PROP_LICENSEE_NAME] = licenseeName;
     }
 
     const String_t& getLicenseeName() const {
-      return licenseeName_i;
+      return licenseeParameters_i.find(PROP_LICENSEE_NAME)->second;
```

Comment:
Also, this code (and similar case below) is not safe with implicit conversion to the return type `String_t&` from `std::string`: a temporary `String_t` object is created, causing returning reference to local. Since `String_t` is not a standard C++, best would be to change call signatures to `std::string`, but this is to be done consistently across the whole codebase.